### PR TITLE
fix(aws-ecs-patterns, aws-elasticloadbalancingv2): Pass TargetGroup P…

### PR DESCRIPTION
**Pull request by yiluncui**
Pass TargetGroup ProtocolVersion as parameters to higher level constructs (#14091)

This change allows users to optionally provide ProtocolVersion when
adding targets to ALB listeners as well as specifying ProtocolVersion
into ApplicationLoadBalancedService patterns in ecs-patterns.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts [hunk 1]
```diff
@@ -363,6 +363,7 @@ export class ApplicationListener extends BaseListener implements IApplicationLis
       healthCheck: props.healthCheck,
       port: props.port,
       protocol: props.protocol,
+      protocolVersion: props.protocolVersion,
```
Comment (njlynch) [inline packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts:366]: What do you think about adding `protocolVersion` to `AddApplicationTargetsProps` as well?

**Review by njlynch — COMMENTED** [review]
Thanks for this contribution! The ELB stuff looks good to me. Pinging @MrArnoldPalmer or @SoManyHs to sign off on the ECS patterns stuff.

File: packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts (hunk 1 above)
Comment (yiluncui) [reply packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts:366]: Are you referring to this?

https://github.com/aws/aws-cdk/pull/14092/files/eff234b0407be3d2f1cb1e8b501b0fa48bf0d056#diff-87cb97a4a8916da7963bded49279e74120b58fd83b956028106159f6518fc761R798-R812

Comment (njlynch) [reply packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts:366]: Oh yeah, sorry, must have mis-read the diff as-is. This looks great then.

**Review by njlynch — APPROVED** [review]

**Pull request merged**